overlays/python-unsupported-platforms.nix

```diff
@@ -0,0 +1,26 @@
+final: prev:
+let
+  inherit (prev) lib;
+  inherit (import ../lib { inherit lib; }) checkBuildPythonPackageFor;
+
+  checkDerivation = drvArgs: drv:
+    let
+      python = lib.lists.last (builtins.filter (p: p ? pythonVersion) drv.propagatedBuildInputs);
+      drvPlatforms = drvArgs.meta.platforms or [ ];
+      unsupportedPlatforms = lib.lists.subtractLists python.meta.platforms drvPlatforms;
+    in lib.singleton {
+      name = "python-unsupported-platforms";
+      cond = (unsupportedPlatforms != [ ]) || (lib.lists.naturalSort drvPlatforms == lib.lists.naturalSort python.meta.platforms);
+      msg = if unsupportedPlatforms != [ ] then ''
+        The following meta.platforms are not supported by the ${python} interpreter:
+          [ ${builtins.toString unsupportedPlatforms} ]
+      '' else ''
+        The meta.platforms attribute is set to the same value by default.
+      '';
+      locations = [
+        (builtins.unsafeGetAttrPos "platforms" python.meta)
```

Comment:
I agree in principle but in this case the benefit of pointing to the relevant source seems worth the tradeoff. It explains the point of the check more efficiently than additional prose would do and tells the user what the default value is which they would otherwise have to hunt down. Here's an example error:

```shell
$ nix run ../nixpkgs-hammering hindsight
warning: Git tree '/home/user/nixpkgs-hammering' is dirty
When evaluating attribute ‘hindsight’:
warning: python-unsupported-platforms
The following meta.platforms are not supported by the /nix/store/2wwb85hrvvfzkhrlwcj3pqi60nlbi9cq-python3-3.8.7 interpreter:
  [ i686-cygwin i686-freebsd i686-netbsd i686-openbsd x86_64-cygwin x86_64-freebsd x86_64-netbsd x86_64-openbsd x86_64-solaris x86_64-windows i686-windows wasm64-wasi wasm32-wasi x86_64-redox arm-none armv6l-none aarch64-none avr-none i686-none x86_64-none powerpc-none msp430-none riscv64-none riscv32-none vc4-none or1k-none mmix-mmixware js-ghcjs aarch64-genode i686-genode x86_64-genode ]

Near /home/user/nixpkgs/pkgs/development/interpreters/python/cpython/default.nix:412:5:
    |
412 |     platforms = with platforms; linux ++ darwin;
    |     ^
Near /home/user/nixpkgs/pkgs/development/tools/hindsight/default.nix:69:5:
   |
69 |     platforms = platforms.all;
   |     ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-unsupported-platforms.md
```

I'm happy to defer to your judgment if you disagree though.